toolchain/check/decl_name_stack.cpp

```diff
@@ -154,20 +154,9 @@ auto DeclNameStack::AddName(NameContext name_context, SemIR::InstId target_id,
           context_->AddExport(target_id);
         }
 
-        auto add_scope = [&] {
-          int index = name_scope.names.size();
-          name_scope.names.push_back(
-              {.name_id = name_context.unresolved_name_id,
-               .inst_id = target_id,
-               .access_kind = access_kind});
-          return index;
-        };
-        auto result = name_scope.name_map.Insert(
-            name_context.unresolved_name_id, add_scope);
-        CARBON_CHECK(
-            result.is_inserted(),
-            "Duplicate names should have been resolved previously: {0} in {1}",
```

Comment:
I was going to approve, but I see the diagnostic is changed. 
The diagnostic on duplicate entry before:
`"Duplicate names should have been resolved previously: {unresolved_name_id} in {parent_scope_id}"`
The diagnostic on duplicate now:
`"Failed to add required name: {unresolved_name_id}"`

Maybe that's fine, or maybe we should pass in the parent scope id, now sure I'll let someone else review.

Reply:
Note that this is a crash diagnostic, so not expected to occur unless there's a bug in the compiler logic.